cmd/kubeadm/app/phases/upgrade/staticpods_test.go

```diff
@@ -193,14 +209,60 @@ func (spm *fakeStaticPodPathManager) BackupEtcdDir() string {
 	return spm.backupEtcdDir
 }
 
+type fakeTLSEtcdClient struct{ TLS bool }
+
+func (c fakeTLSEtcdClient) HasTLS() bool {
+	return c.TLS
+}
+
+func (c fakeTLSEtcdClient) GetStatus() (*clientv3.StatusResponse, error) {
```

Comment:
same as other comment, I'm an anti-stub person and we have the jiggery to dynamically spin clusters in UTs.

Reply:
@timothysc this sounds great for testing `GenericClient.GetStatus()` in `util/etcd/etcd_test.go`

Do we have an existing unit-test harness for running etcd as a static pod using a kubelet?
That would be what is required for this particular test.

Reply:
We have utilities to basically spin up an actual etcd server + client pair, but at this point I'm ok with merging what is here and we can create a new issue and cross-link for future work.